hypershift-operator/controllers/nodepool/kubevirt.go

```diff
@@ -161,9 +161,18 @@ func virtualMachineTemplateBase(image string, kvPlatform *hyperv1.KubevirtNodePo
 	}
 
 	if kvPlatform.RootVolume != nil &&
-		kvPlatform.RootVolume.Persistent != nil &&
-		kvPlatform.RootVolume.Persistent.StorageClass != nil {
-		dataVolume.Spec.Storage.StorageClassName = kvPlatform.RootVolume.Persistent.StorageClass
+		kvPlatform.RootVolume.Persistent != nil {
+		if kvPlatform.RootVolume.Persistent.StorageClass != nil {
+			dataVolume.Spec.Storage.StorageClassName = kvPlatform.RootVolume.Persistent.StorageClass
+		}
+		if kvPlatform.RootVolume.Persistent.AccessModes != nil {
```

Comment:
shouldn't  this be something like...

```
if len(kvPlatform.RootVolume.Persistent.AccessModes) != 0 {
```

otherwise i'm pretty sure this condition will always `!= nil` which means the if statement wouldn't really be necessary.

Reply:
Right, because it is an array so it can't be nil.
Fixing, thanks.

Reply:
Done.
Also I've noticed there was already a code block that populates the `dataVolume` object from `kvPlatform`, so I unified them into one block.